tests/testsuite/git_status.rs

```diff
@@ -5,6 +5,14 @@ use std::process::Command;
 
 use crate::common::{self, TestCommand};
 
+/// Ensures commands finish on windows
+#[cfg(not(windows))]
+pub fn barrier() {}
+#[cfg(windows)]
+pub fn barrier() {
+    std::thread::sleep(std::time::Duration::from_millis(500));
```

Comment:
This feels like a hack, why does it work and what is it solving?

Also, regarding the comment, is it _Windows_ specific or is it _PowerShell_ specific?

Reply:
Agree, I would like to understand this better.

Reply:
I looks like right after the calls to git the filesystem state hasn't finished updating yet causing some of the tests to fail ([test run before change](https://github.com/starship/starship/runs/249566711), [test run after change](https://github.com/starship/starship/runs/249783836)). I also added these barriers after each call to git, even where it hadn't caused issues yet, to reduce possible test flakiness. I believe it isn't trivial to start a full fs sync on windows.
This addition is Windows specific though other operating systems could have the same issue. None of the tests cover anything powershell-specific yet.

Reply:
Ah okay, good to know thanks!